lib/sqlalchemy/sql/elements.py

```diff
@@ -2766,6 +2766,9 @@ def _construct_raw(
         self._is_implicitly_boolean = False
         return self
 
+    def __getitem__(self, item) -> ColumnElement[Any]:
+        return self.clauses[item]
```

Comment:
I'm not 100% sure this is the best way.

maybe we can do something like
```
clauses = [elem._compiler_dispatch(self, **kw) for elem in fn.clauses]
args = ', '.join(clauses[:-1])
sep = caluses[-1]
```

Reply:
good suggestion, updated